arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -447,3 +447,9 @@ inline fun <E, A, B> NonEmptyList<A>.traverseValidated(
 
 fun <E, A> NonEmptyList<Validated<E, A>>.sequenceValidated(semigroup: Semigroup<E>): Validated<E, NonEmptyList<A>> =
   traverseValidated(semigroup, ::identity)
+
+inline fun <A, B> NonEmptyList<A>.traverseOption(f: (A) -> Option<B>): Option<NonEmptyList<B>> =
+  traverseEither { f(it).toEither { Unit } }.orNone()
```

Comment:
```suggestion
inline fun <A, B> NonEmptyList<A>.traverseOption(f: (A) -> Option<B>): Option<NonEmptyList<B>> {
  val acc = mutableListOf<B>()
  forEach { a ->
    when (val res = f(a)) {
      is Right -> acc.add(res.value)
      is Left -> return@traverseOption res
    }
  }
  // Safe due to traverse laws
  return NonEmptyList.fromListUnsafe(acc).right()
}
```

Nit: a direct implementation reads better IMO, and is (slightly) more performant.
That'll save 1 allocation per element in the NonEmptyList.